wallet/test/Test/Pos/Wallet/Web/Methods/BackupDefaultAddressesSpec.hs

```diff
@@ -0,0 +1,87 @@
+{-# OPTIONS_GHC -fno-warn-unused-imports #-}
+{-# OPTIONS_GHC -fno-warn-unused-matches #-}
+{-# OPTIONS_GHC -fno-warn-unused-local-binds #-}
+module Test.Pos.Wallet.Web.Methods.BackupDefaultAddressesSpec
+       ( spec
+       ) where
+
+import           Universum
+
+import           Data.Default (def)
+import           Formatting ()
+import           Pos.Crypto (firstHardened)
+import           Pos.Crypto.Signing.Types.Safe (emptyPassphrase)
+import           Pos.Launcher (HasConfigurations)
+import           Pos.Util.BackupPhrase (mkBackupPhrase9, bpToList)
+import           Pos.Util.CompileInfo (HasCompileInfo, withCompileInfo)
+import           Pos.Wallet.Web.Account (GenSeed(..))
+import           Pos.Wallet.Web.Backup (WalletBackup(..), getWalletBackup)
+import           Pos.Wallet.Web.ClientTypes (CWallet(..), CWalletMeta(..),
+                                             CWalletInit(..), CWalletAssurance(..),
+                                             Wal(..), CId(..), CHash(..), encToCId)
+import           Pos.Wallet.Web.Methods.Backup (restoreWalletFromBackup)
+import           qualified Pos.Wallet.Web.Methods.Logic as L
+import           Pos.Wallet.Web.Methods.Restore (newWallet)
+import           Pos.Wallet.Web.State (AddressLookupMode(Ever), getAccountWAddresses,
+                                       getWalletMeta, getWalletAddresses)
+import           Pos.Wallet.Web.Util (getWalletAccountIds)
+import           Test.Hspec (Spec, describe)
+import           Test.Pos.Util (assertProperty, stopProperty, expectedOne,
+                                withDefConfigurations)
+
+import Test.Pos.Wallet.Web.Mode (walletPropertySpec)
+
+
+spec :: Spec
+spec = withCompileInfo def $
+       withDefConfigurations $
+       describe "restoreAddressFromWalletBackup" $ restoreWalletAddressFromBackupSpec
+
+restoreWalletAddressFromBackupSpec :: (HasCompileInfo, HasConfigurations) => Spec
+restoreWalletAddressFromBackupSpec = walletPropertySpec restoreWalletAddressFromBackupDesc $ do
+    -- I NEED FIRST TO CREATE WALLETBACKUP -> HOW ?
+    -- MAYBE LIKE THIS
+    -- WALLETBACKUP CREATION
+    -- wBackup <- lift $ getWalletBackup ((CId (CHash "test")) :: CId Wal)
+
+    -- OR CREATE NEW WALLET ?
```

Comment:
I believe you should (as you did) create an *arbitrary* wallet, export it and reimport it. Then test did you get object with same properties back. Pseudo:
```
do
  wallet <- pick $ arbitrary -- this is pseudo. Read bellow for more details
  filePath <- "some filepath that works on Win/Linux"
  exportWalletJSON wallet filePath
  walletImp <- importWalletJSON filePath
  -- test various properties expected of the two. Also test is imported thing the same as the one we exported with Eq
```

For how to create an arbitrary wallet you can refer to https://github.com/input-output-hk/cardano-sl/blob/master/wallet/test/Test/Pos/Wallet/Web/Methods/PaymentSpec.hs#L47 `importSomeWallets` and then picking an arbitrary wallet from available wallets with https://github.com/input-output-hk/cardano-sl/blob/master/wallet/test/Test/Pos/Wallet/Web/Methods/PaymentSpec.hs#L51 : 
```
    rootsWIds <- lift myRootAddresses
    idx <- pick $ choose (0, l - 1)
    let walId = rootsWIds !! idx
let pswd = passphrases !! idx
```

then you can export this wallet ^ and do the rest of tests.